### ISSUE TITLE
Location of banner field text

### ISSUE THREAD

**Issue by davidspring** [friendica/friendica#13184]
To change the site banner, one goes to Admin, Site, General Information where there is a Banner Logo box which has the following text: 
```
<a href="https://friendi.ca/"><img id="logo-img" width="32" height="32" src="images/friendica-banner.jpg" alt="logo" /></a><span id="logo-text"><a href="https://friendi.ca">Friendica</a></span>
```
I would like to change some of this text above. I am starting to figure out the Friendica template system. So I went to view/templates/admin where there is a template called 
site.tpl. 

On Line 19 of this admin site template is this: 
{{include file="field_textarea.tpl" field=$banner}}

I then found a template called field_textarea.tpl. But it just gave general instructions for any text area field. So I am wondering where I can find the **field=$banner** which I assume will specify the text above. 

Thanks again for your help and patience.

**Comment by MrPetovan** [comment]
Here's where the `$banner` template variable is defined:
https://github.com/friendica/friendica/blob/2fc7c9f064bfcfe8302886bc479c8ab5b4e06c8c/src/Module/Admin/Site.php#L423

**Comment by davidspring** [comment]
I was looking at this file earlier today and I just reread it twice. All I see for Banner is Line 62: 
$banner           = (!empty($_POST['banner'])           ? trim($_POST['banner'])                             : false);

and Line 176
if ($banner == "") {$transactionConfig->delete('system', 'banner');
		} else {
	$transactionConfig->set('system', 'banner', $banner);}

and Line 360 
$banner = DI::config()->get('system', 'banner');

and Line 423: 
'$banner'           => ['banner', DI::l10n()->t('Banner/Logo'), $banner, ''],

Line 423 does not seem to me to be where the text in the banner box is set. Here again is the text I am trying to change: 

`<a href="domain-url"><img id="logo-img" width="32" height="32" src="images/custom-banner.jpg" alt="logo" /></a><span id="logo-text"><a href="domain-url">Domain Name</a></span>`

In particular, I want to change the src= folder path and file path. This path must be specified someplace. Any ideas?

**Comment by MrPetovan** [comment]
The default for this config value is defined here:
https://github.com/friendica/friendica/blob/2fc7c9f064bfcfe8302886bc479c8ab5b4e06c8c/static/settings.config.php#L65

**Comment by davidspring** [comment]
Thank you. That was the Line I was looking for. I would like to change the path used by src= to

“view/theme/(specify theme)/images/banner.jpg”

For example, I have created an images folder in the vier theme folder and added the banner image to that. I realize I can just define the path in the box to the vier theme. But I would like to find a way so that when a person is picking out a theme, they are automatically using the banner associated with that theme. 

Is there a way to use a theme variable in this path so that each theme could have its own banner image?

**Comment by davidspring** [comment]
I have managed to solve the problem. I first needed to delete most of Line 65 in settings.config.php as pointed out above leaving only the banner but deleting the path in the parathesis to leave it empty 
I was then able to delete the text in the admin/site banner/logo box and have the box stay empty when I saved the setting. 

This obviously created problems for displaying a banner image using the #banner. However, I was able to use CSS on the header element which contains the banner. By putting the CSS for the header element at the end of the variant CSS files, I was able to route different banner (header) images to different theme variants. 
So I now have a theme folder with an images folder that has three different background images for four different color variants. It looks pretty spectacular – way better than Facebook. 

I still have a lot of work left. But I should be able to submit the new theme for your review some time next week.
